src/foraging_gui/Foraging.py

```diff
@@ -2307,7 +2333,22 @@ def _StartTrialLoop(self,GeneratedTrials,worker1):
                     # get reward and generate a new trial
                     self.NewTrialRewardOrder=0     
                 #initiate the generated trial
-                GeneratedTrials._InitiateATrial(self.Channel,self.Channel4)
+                try:
+                    GeneratedTrials._InitiateATrial(self.Channel,self.Channel4)
+                except Exception as e:
```

Comment:
There are other possibilities that you got an error from GeneratedTrials._InitiateATrial(self.Channel,self.Channel4)

Reply:
You are correct! I've updated this so it now checks for "ConnectionAbortedError".